File: 2021년/9월/0924 운동.py
# 백준_1956_운동_플로이드-와샬_골드 4

import sys
input = sys.stdin.readline
INF = sys.maxsize

v, e = map(int, input().split())
city = [[INF for _ in range(v)] for _ in range(v)]
for i in range(e):
    a, b, c = map(int, input().split())
    city[a-1][b-1] = c

for k in range(v):
    for i in range(v):
        for j in range(v):
            city[i][j] = min(city[i][j], city[i][k] + city[k][j])

min_num = INF
for i in range(v):
    if city[i][i] == INF:
        continue
    else:
        min_num = min(min_num, city[i][i])

if min_num == INF:
    print(-1)
else:
    print(min_num)

File: 2021년/9월/0924 운동.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int v,e;
int city[400][400];
int INF = 1000000000;

int main()
{
	scanf("%d %d", &v, &e);
	
	for(int i=0;i<v;i++)
	{
		for(int j=0;j<v;j++) city[i][j] = INF;
	}
	
	for(int i=0;i<e;i++)
	{
		int a,b,c;
		scanf("%d %d %d", &a, &b, &c);
		city[a-1][b-1] = c;
	}
	
	for(int k=0;k<v;k++)
	{
		for(int i=0;i<v;i++)
		{
			for(int j=0;j<v;j++) city[i][j] = min(city[i][j], city[i][k]+city[k][j]);
		}
	}
	
	int result = INF;
	for(int i=0;i<v;i++)
	{
		if(city[i][i] == INF) continue;
		else result = min(result, city[i][i]);
	}
	
	if(result == INF) printf("%d",-1);
	else printf("%d", result);
}
